core/core-impl/src/main/java/ai/timefold/solver/core/config/solver/SolverJobConfig.java

```diff
@@ -0,0 +1,101 @@
+package ai.timefold.solver.core.config.solver;
+
+import java.util.UUID;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+import ai.timefold.solver.core.config.solver.termination.TerminationConfig;
+
+/**
+ * Settings used by {@link ai.timefold.solver.core.api.solver.SolverManager} to submit and solve problems. It also
+ * includes settings to override default {@link ai.timefold.solver.core.api.solver.Solver} configuration.
+ *
+ * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+ * @param <ProblemId_> the ID type of a submitted problem, such as {@link Long} or {@link UUID}.
+ */
+public class SolverJobConfig<Solution_, ProblemId_> {
```

Comment:
I understand why you decided to introduce it, but from the point of view of the user, this is boilerplate.
If you want to keep this class, please move it outside of `config`/`api` packages, where it becomes public API, and make sure public API does not reference it in any API contract.

Reply:
I removed this class as it had become unused after updating the public API contract.